scripts/commit_pr.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# ASSUMPTIONS:
+# - You are in project root
+# - There are changes to commit
+# - You are ready to make a PR
+# - You are logged in via CF CLI 
+# - Your branch contains the Jira issue number e.g. SC-123
+
+# USAGE:
+#   ./scripts/commit_pr.sh "Commit message"
+
+set -euo pipefail
+
+COMMIT_MSG=$1
+BRANCH=`git rev-parse --abbrev-ref HEAD`
+URL_BRANCH=`echo $BRANCH | sed 's/\//%2F/g'`
+JIRA_PATTERN='sc-[[:digit:]]*'
+JIRA_CODE=`echo $BRANCH | grep -oi $JIRA_PATTERN`
+SUBDOMAIN="smart-cities-$JIRA_CODE"
+INSTANCE="http://$SUBDOMAIN.apps.staging.digital.gov.au"
+
+git add -A
+git commit -m "$COMMIT_MSG"
+git push 
+
+npm run build 
+cf push $SUBDOMAIN -c "null"
+
+open "https://github.com/govau/smart-cities/compare/$URL_BRANCH?expand=1&body=Preview:+<$INSTANCE>"
```

Comment:
Is it worth having a note in the readme about what this is?

Reply:
I am actually going to refactor this script in a separate branch, at which point I'll document it ... I'm going to give it opts so it can suit a bunch of cases (not just the assumptions at top of script) and also provide teardown. 

This might end up as something sharable with other projects that have a similar workflow.